src/oid4vp/holder.rs

```diff
@@ -0,0 +1,511 @@
+        .map_err(|e: anyhow::Error| OID4VPError::PresentationSubmissionCreation(e.to_string()))?;
+
+        let vp_token = self
+            .create_unencoded_verifiable_presentation(&request, &response.selected_credential)
+            .await?;
+
+        let response = self
+            .submit_response(
+                request.to_owned(),
+                AuthorizationResponse::Unencoded(UnencodedAuthorizationResponse(
+                    Default::default(),
+                    vp_token,
+                    presentation_submission,
+                )),
+            )
+            .await
+            .map_err(|e| OID4VPError::ResponseSubmission(e.to_string()))?;
+
+        // Reset the authorization request after the response has been submitted.
+        self.authorization_request.write().await.take();
+
+        Ok(response)
+    }
+}
+
+// Internal methods for the Holder.
+impl Holder {
+    /// This will return all the credentials that match the presentation definition.
+    async fn search_credentials_vs_presentation_definition(
+        &self,
+        definition: &PresentationDefinition,
+    ) -> Result<Vec<Arc<ParsedCredential>>, OID4VPError> {
+        let credentials = match &self.provided_credentials {
+            // Use a pre-selected list of credentials if provided.
+            Some(credentials) => credentials.to_owned(),
+            None => match &self.vdc_collection {
+                None => vec![],
+                Some(vdc_collection) => {
+                    let credential_ids = vdc_collection.all_entries().await?;
+
+                    futures_util::stream::iter(credential_ids)
+                        .filter_map(|id| async move { vdc_collection.get(id).await.ok().flatten() })
+                        .collect::<Vec<Arc<Credential>>>()
+                        .await
+                }
+            },
+        }
+        .into_iter()
+        .filter_map(|cred| cred.try_into_parsed().ok())
+        .filter_map(
+            |cred| match cred.check_presentation_definition(definition) {
+                true => Some(cred),
+                false => None,
+            },
+        )
+        .collect::<Vec<Arc<ParsedCredential>>>();
+
+        Ok(credentials)
+    }
+
+    // Construct a DescriptorMap for the presentation submission based on the
+    // credentials returned from the VDC collection.
+    fn create_descriptor_map(
+        &self,
+        credential: &Arc<ParsedCredential>,
+        input_descriptor_id: String,
+    ) -> Result<DescriptorMap, OID4VPError> {
+        Ok(DescriptorMap::new(
+            input_descriptor_id,
+            ClaimFormatDesignation::JwtVpJson,
+            "$".into(),
+        )
+        // Credentials will be nested within a `vp` JSON object.
+        .set_path_nested(DescriptorMap::new(
+            credential.id().to_string(),
+            ClaimFormatDesignation::from(credential.format()),
+            "$.verifiableCredential".into(),
+        )))
+    }
```

Comment:
This needs a lot more work, you are only handling a particular credential format here. Need to match on the format in the presentation definition and the format of the selected credential.

Reply:
Alternatively, only add support for vcdm2.0_sd_jwt for now.

Reply:
Yes, you're right. This was a hold over from misunderstanding around the shape of the vp token, thinking it was itself a claim (JWT) with an inner verifiableCredential field.

Since the vp token is not itself a verifiable claim, I've removed the nested path and now use the format of the credential at the top level. 

We can amend this in the future when we support multi-claim values. I'll open a ticket for it.